File: GestionEventosDeportivosBackend/Models/InscripcionModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GestionEventosDeportivos.Models
{
    public class InscripcionModel
    {
        [Key]
        public int inscripcion_id { get; set; }

        [Required]
        public int evento_id { get; set; }

        [Required]
        public int participante_id { get; set; }

        [Required]
        public DateTime fecha_inscripcion { get; set; } = DateTime.Now;

        [StringLength(20)]
        public string estado { get; set; } = "Activa";

        // Relaciones
        [ForeignKey("evento_id")]
        public virtual EventoModel Evento { get; set; }

        [ForeignKey("participante_id")]
        public virtual ParticipanteModel Participante { get; set; }
    }
}

File: GestionEventosDeportivosFrontend/src/app/models/inscripcion.model.ts
export interface Inscripcion {
  inscripcion_id: number;
  evento_id: number;
  participante_id: number;
  fecha_inscripcion: Date;
  estado: string;
  evento_nombre: string;
  participante_nombre_completo: string;
}

export interface CreateInscripcion {
  evento_id: number;
  participante_id: number;
}

export interface UpdateInscripcion {
  estado: string;
}